inductiva/templating/renderers.py

```diff
@@ -0,0 +1,120 @@
+"""Base and specific renderers packages to manage template files."""
+from abc import abstractmethod
+from typing import Iterable, Union, IO
+import pathlib
+import shutil
+import os
+
+import jinja2
+
+from inductiva import types, utils
+
+
+class BaseRenderer:
+    """Abstract base class for template renderers."""
+
+    TEMPLATE_EXTENSION = None
+
+    @abstractmethod
+    def render_dir(self, target_dir: types.Path, source_dir: types.Path,
+                   **render_args):
+        """Render a template to a file."""
+        pass
+
+    @abstractmethod
+    def render_file(self, target_file: types.Path,
+                    source_file: Union[types.Path, IO], **render_args):
+        """Render a file."""
+        pass
+
+    @classmethod
+    def is_template(cls, file: types.Path) -> bool:
+        """Check if the given file is of template type."""
+        if isinstance(file, pathlib.Path):
+            return file.suffix == cls.TEMPLATE_EXTENSION
```

Comment:
Not sure how this works with `cls.TEMPLATE_EXTENSION` that start with a dot (ex. `.jinja`)

Reply:
This works!